### ISSUE TITLE
[Bug]: Why does the validateMainThread run sync on the main thread?

### ISSUE THREAD

**Issue by azilbershtein** [braze-inc/braze-swift-sdk#89]
### Platform

iOS

### Platform Version

all

### Braze SDK Version

7.1.0

### Xcode Version

al

### Computer Processor

Intel

### Repro Rate

100%

### Steps To Reproduce

On BrazeInAppMessageUI.swift there's a validateMainThread function, why does it run synchronize on the main thread in order to send log? theoretically it can create unusual behaviors on the app.
Can you change it to async? or not the main thread at all if it's just to send log.

The function:
`  func validateMainThread(for message: Braze.InAppMessage) -> Bool {
    guard Thread.isMainThread else {
      DispatchQueue.main.sync {
        logError(for: message.context, error: .noMainThread)
      }
      return false
    }
    return true
  }`

### Expected Behavior

Should not use the main thread/ sync

### Actual Incorrect Behavior

Run sync on the main thread

**Comment by jacksonemiller** [comment]
Hi @azilbershtein, thank you for reaching out. This function does enqueue work on to the main thread. However, as a result of [the guard statement](https://github.com/braze-inc/braze-swift-sdk/blob/d2b2407bb184e1eded6209458be94e7761ed4b91/Sources/BrazeUI/InAppMessageUI/InAppMessageUI.swift#L282), it could only ever block a background thread and not the main thread as it waited for that work to be completed. Is blocking the main thread primarily what you were concerned about? Thanks.

**Comment by azilbershtein** [comment]
Hi @jacksonemiller
Thanks for your answer.
The guard statement basically make sure this function run on the main thread, but let's say it doesn't, the background thread will be blocked by DispatchQueue.main.sync, but you don't know what the main is doing at this point, so it can be blocked by another task that might be blocked by the same background task you're running.
Basically it's just not good practice to block the main from a background thread :) (at least not for a log).
Any chance you can just change it to async instead of sync?

**Comment by jacksonemiller** [comment]
Ah I think I understand the situation you're worried about now. Makes sense, thanks for pointing this out. I've filed a ticket to implement the change, and it should be released as part of the changes in our next release. Thanks!

**Comment by jacksonemiller** [comment]
Hi again @azilbershtein,

We just released a fix for this as part of our 7.2.0 release. Thanks!
